### PR TITLE
Add new cleanup-tool for repository files

### PR DESCRIPTION
**Pull request by ManInFez**
resolves equinor/komodo-releases#103

### CODE REVIEW

**Comment by anders-kiaer** [comment]
If we are going into the direction of removing comments from `repository.yml`, we can remove the `ruamel.yaml` dependency later perhaps also for `komodo-prettier` 🙂 

If we want to create a smaller `repository.yml` (i.e. remove unused versions/packages), and _keep_ comments for the remaining parts of the repository `.yml` file, we could probably reuse most of `komodo-prettier` code and e.g. add a flag `--cleanrepository` to the existing `komodo-prettier` entrypoint.

**Comment by ManInFez** [comment]
Yes it might be a good idea to combine prettier with cleanup to preserve comments.

File: komodo/release_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        if len(repository[lib]) == 0:
+            repository.pop(lib)
+
+
+def _sorting_versions(item):
+    # put master first
+    if "master" in item[0]:
+        return [0]
+
+    version_numbers = re.sub("[^0-9\.]", "", item[0])
```
Comment (oysteoh) [inline komodo/release_cleanup.py]: to avoid `DeprecationWarning: invalid escape sequence \.` - use r'[^0-9\.]'

File: komodo/release_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        output_file = args.output
+        write_to_file(args.repository, output_file)
+        if unused_versions: 
+            print("Success! New repository file written to {}".format(output_file))
+            print("The following software version are not in use:")
+            for lib, versions in unused_versions.items():
+                print(lib, versions)
+
+
+def run_prettier(args, parser):
```
Comment (oysteoh) [inline komodo/release_cleanup.py]: parser not used

File: komodo/release_cleanup.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
+import os
+import sys
+import ruamel
+import yaml
+import textwrap
+import re
+import functools
+
```
Comment (oysteoh) [inline komodo/release_cleanup.py]: ruamel, yaml, textwrap and functools not used

File: komodo/release_cleanup.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+from argparse import ArgumentError, ArgumentParser, ArgumentTypeError
+from collections import OrderedDict
+from komodo import prettier, load_yaml, write_to_file, prettified_yaml
+
```
Comment (oysteoh) [inline komodo/release_cleanup.py]: ruamel, yaml, textwrap, functools, OrderedDict and prettier not used

File: komodo/release_cleanup.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        type=load_yaml,
+        help="a yml file defining all libraries and their versions",
+    )
+    cleanup_parser.add_argument(
+        "--releases",
+        type=_valid_path_or_files,
+        help="list of release files or folders containing releases",
+        nargs="+",
+    )
+    cleanup_parser.add_argument(
```
Comment (jondequinor) [inline komodo/release_cleanup.py]: Please output `--output` to `STDOUT` if not specified.

And output from `--check` should be suppressed if not explicitly asked for? If I provide `--output` I still get `--check` output on `STDOUT`.

File: tests/data/test_releases/2020.01.a1-py27.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+lib1: 0.1.2
+lib2: 1.2.3
+lib3: 2.3.4
+lib4: 3.4.5
```
Comment (sondreso) [inline tests/data/test_releases/2020.01.a1-py27.yml]: Missing newline :male_detective:

File: tests/data/test_repository.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  3.4.5:
+    depends:
+    - lib2
+    - lib3
+
+lib5: # comment should be gone
+  1.2.3:
+    depends:
+    - lib2
+    - lib3
```
Comment (sondreso) [inline tests/data/test_repository.yml]: Missing newline :male_detective:

Comment (ManInFez) [reply tests/data/test_repository.yml]: The cleanup tool fixes this ;)

Comment (sondreso) [reply tests/data/test_repository.yml]: Nice! :D

**Comment by jondequinor** [comment]
Could you move the formatting changes into its own commit? And maybe clean up the commits a bit?

File: komodo/release_cleanup.py (hunk 5 above)
Comment (ManInFez) [reply komodo/release_cleanup.py:108]: stdout option added

File: tests/__init__.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+def _load_yaml(filename):
+    ruamel_instance = ruamel.yaml.YAML()
+    ruamel_instance.indent(  # Komodo prefers two space indendation
+        mapping=2, sequence=4, offset=2
+    )
+    ruamel_instance.width = 1000  # Avoid ruamel wrapping long
+    with open(filename) as repo_handle:
+        input_dict = ruamel_instance.load(repo_handle)
+    return input_dict
```
Comment (jondequinor) [inline tests/__init__.py:15]: no newline in end of file

Comment (ManInFez) [reply tests/__init__.py:15]: :+1:

**Pull request merged**